client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js

```diff
@@ -51,10 +49,13 @@ export const Sidebar = props => {
 	} = props;
 
 	const onFulfillAndEmailOrderChange = ( value ) => {
-		// Don't change order status if already finished.
-		props.setFulfillOrderOption( orderId, siteId, value && ! isOrderFinished( order.status ) );
-		// Email only if order is already complete.
-		props.setEmailDetailsOption( orderId, siteId, value && isOrderFinished( order.status ) );
+		if ( isOrderFinished( order.status ) ) {
+			// Email only if order is already complete.
+			props.setEmailDetailsOption( orderId, siteId, value );
+		} else {
+			// Don't change order status if already finished.
+			props.setFulfillOrderOption( orderId, siteId, value );
+		}
```

Comment:
Instead of updating both all the time, now it updates either or based on `order.status`. I think this is fine since we have defaults and we don't need to update both every time.